include/boost/math/ccmath/sqrt.hpp

```diff
@@ -0,0 +1,56 @@
+//  (C) Copyright Matt Borland 2021.
+//  Use, modification and distribution are subject to the
+//  Boost Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+//  Constexpr implementation of sqrt function
+
+#ifndef BOOST_MATH_CCMATH_SQRT
+#define BOOST_MATH_CCMATH_SQRT
+
+#include <limits>
+#include <type_traits>
+#include <boost/math/ccmath/isnan.hpp>
+#include <boost/math/ccmath/isinf.hpp>
+
+namespace boost { namespace math { namespace ccmath { 
```

Comment:
I'd do `namespace boost::math::ccmath` just so if you need to make backwards incompatible changes in C++14 mode you don't surprise anyone.